packages/amplify_datastore/ios/Classes/types/model/FlutterModelFieldType.swift

```diff
@@ -0,0 +1,66 @@
+/*
+* Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+
+import Foundation
+import Amplify
+
+public struct FlutterModelFieldType {
+    public let fieldType : String
+    public let ofModelName : String?
```

Comment:
Can we use enum here similar to `ModelFieldType`

Reply:
Hmm I don't see the benefit to that approach.  This class is meant to take a model and then generate the Amplify-iOS ModelFieldType enum afterwards.  I don't see the benefit of creating yet another enum that will just be converted to the final enum right after

Reply:
By using enum, you can avoid the optional `ofModelName:String?` and assign an associated value of `String` only for `model` and `collection`. This will avoid force casting or nil check later.

But if this class is just storing the json data, I think it is okay to keep like this.

Reply:
cool I'll keep it as is for now then